### ISSUE TITLE
Incorrect total size in search result

### ISSUE THREAD

**Issue by kyriiherman** [webpack-contrib/webpack-bundle-analyzer#251]
#### Issue description
When I search for some modules total size isn't correct. It looks like total size is folder size + size of each file in folder

#### Technical info
* Webpack Bundle Analyzer version: 3.0.3
* Webpack version: 4.19.1
* Node.js version: 11.9.0
* npm/yarn version: 1.13.0
* OS: macOS Mojave

#### Debug info
How do you use this module? As CLI utility or as plugin?
plugin

If plugin, what options were provided? (e.g. `new BundleAnalyzerPlugin({ analyzerMode: 'disabled', generateStatsFile: true })`)
{ openAnalyzer: false }

<img width="403" alt="screen shot 2019-02-14 at 13 04 00" src="https://user-images.githubusercontent.com/9069967/52783881-fdd0fd80-305b-11e9-93f7-22094c954de2.png">

**Comment by valscion** [comment]
Hmm are you sure your lodash in `5.chunk.js` isn't that large? Does it show a different size in the treeview?

**Comment by th0r** [comment]
I think its a legit bug. I've overlooked this case.

**Comment by kyriiherman** [comment]
@th0r @valscion I think bug is here https://github.com/webpack-contrib/webpack-bundle-analyzer/blob/d7682a655abda824c82b9d6bb7991b9576c33610/client/store.js#L153. You should check for `!module.groups` before adding module size

**Comment by th0r** [comment]
@kyriiherman it's not so easy. What if the only found item is a directory? In your case the size will be zero.

**Comment by kyriiherman** [comment]
@th0r yes, this is correct behavior. in your case the directory is empty, otherwise any file inside directory are present in found modules, because files paths include directory name

**Comment by th0r** [comment]
@kyriiherman No, it's not. You can input a regexp that will match only directory name e.g. `node_modules$`.

**Comment by kyriiherman** [comment]
@th0r oh, my fault( so we should check for `this.searchQueryRegexp` and if someone is looking for a directory then we should consider directory size, otherwise no 😏

**Comment by th0r** [comment]
I don't think it will work either. The proper solution would be to check nesting of the found items e.g. if we have found "lodash/find.js" module then we should ignore it's size if we also found a whole "lodash" directory. So this task is not so simple and may require quite significant changes in the data structure that currently represents modules tree.

**Comment by kyriiherman** [comment]
@th0r why you think solution with checking `this.searchQueryRegexp` isn't ok?

**Comment by th0r** [comment]
@kyriiherman `(lodash$)|(react$)` is a valid RegExp that will match either `react` or `lodash` folders, but won't pass your check.

**Comment by kyriiherman** [comment]
@th0r so we need to filter modules tree in `foundModulesByChunk`, but save its structure and then in `foundModules` use `walkModules` to convert modules tree into collection. with this approach we can use tree structure to count `totalSize`

**Comment by anatoliyarkhipov** [comment]
Just to clarify, the problem is still here, right? If yes, what do you think about this regexp, do I get the real number this way? 🤔
![image](https://user-images.githubusercontent.com/1934180/67563082-a7f34000-f728-11e9-946b-9e73c598d8bd.png)

**Comment by valscion** [comment]
Yes the problem is still there. I don't know the answer to your other question, maybe someone else knows.

**Comment by ppJuan** [comment]
same problem
![1610438777186-image](https://user-images.githubusercontent.com/13941067/104286800-84184780-54f0-11eb-8ce5-4d0cb27ada35.png)
